From a0fb1bf5a2b05f75981752ff0c20bf6e450d02c3 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Fri, 7 Aug 2020 15:18:01 +0000
Subject: [PATCH 05/31] libserver: wrap pthread_getname_np

pthread_getname_np is supposed to be wrapped in ifdef, just like
in common/ECLogger.cpp.

References: KF-3318
---
 provider/libserver/ECDatabaseFactory.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/provider/libserver/ECDatabaseFactory.cpp b/provider/libserver/ECDatabaseFactory.cpp
index 6f58c2b0c..b22d1f134 100644
--- a/provider/libserver/ECDatabaseFactory.cpp
+++ b/provider/libserver/ECDatabaseFactory.cpp
@@ -26,7 +26,9 @@ ECDatabaseFactory::ECDatabaseFactory(std::shared_ptr<ECConfig> c,
 		if (arg == nullptr)
 			return;
 		char name[32];
+#ifdef HAVE_PTHREAD_GETNAME_NP
 		pthread_getname_np(pthread_self(), name, sizeof(name));
+#endif
 		ec_log_debug("db_conn %p was not released on T%lu (%s)", arg, kc_threadid(), name);
 	});
 }
-- 
2.29.2

